codegen/introspect_schema.py

```diff
@@ -0,0 +1,53 @@
+from collections.abc import Iterator
+from importlib import import_module
+from pathlib import Path
+from pkgutil import walk_packages
+from types import ModuleType
+from typing import Final
+
+import kio.schema
+
+from kio.static.constants import EntityType
+from kio.static.protocol import Entity
+
+base_dir: Final = Path(__file__).parent.parent.resolve()
+schema_src_dir: Final = Path(kio.schema.__file__).parent.resolve()
+ignore_modules: Final = frozenset({"index"})
+
+
+def generate_modules(parent: ModuleType) -> Iterator[ModuleType]:
+    for package in walk_packages(parent.__path__):
+        if package.name in ignore_modules:
+            continue
+        module = import_module(f"{parent.__name__}.{package.name}")
+        if package.ispkg:
+            yield from generate_modules(module)
+        else:
+            yield module
+
+
+def get_entities() -> Iterator[tuple[type[Entity], str]]:
+    modules = list(generate_modules(import_module("kio.schema")))
+    for module in modules:
+        items = module.__dict__.copy()
+        # Eliminate non-entity modules, situated directly under `kio.schema`.
+        if sum(1 for c in module.__name__ if c == ".") < 3:
```

Comment:
Could be simplified 
```suggestion
        if module.__name__.count(".") < 3:
```

Reply:
Oh nice, I didn't know about `str.count()` 👍 This function is only moved in this PR, extracted from `.generate_tests` to be able to share it with `.generate_index`, and I think it's valuable to not alter it in the same change-set. But I'm happy to incorporate this change in a follow-up if that seems OK?